### ISSUE TITLE
resolvePluginsRelativeTo to resolve extentions

### ISSUE THREAD

**Issue by flyyang** [eslint/eslint#12450]
**Tell us about your environment**

Environment Info:

Node version: v10.15.3
npm version: v6.12.0
Local ESLint version: v6.5.1(currently used)
Global ESLint version: v5.15.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
{
    extends: [
      'airbnb-base',
      'plugin:vue/essential',
    ],
    parserOptions: {
      parser: 'babel-eslint',
      allowImportExportEverywhere: true,
      sourceType: 'module',
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
resolvePluginsRelativeTo: 'somePath'
```

**What did you expect to happen?**

I have a global cli tool like vue cli, which have share config in diffrent folder with the project folder.
What i expect is to resolve plugins in the cli global node_modules.

**What actually happened? Please include the actual, raw output from ESLint.**

Event set resolvePluginsRelativeTo to the global path, eslint still try to find extentions in current working direcotry.

**Are you willing to submit a pull request to fix this bug?**
yes.

I know this topic been talked a lot. But seems a minor change will solve my problem.

[https://github.com/eslint/eslint/blob/master/lib/cli-engine/config-array-factory.js#L690-L691](https://github.com/eslint/eslint/blob/master/lib/cli-engine/config-array-factory.js#L690-L691)

```
        const { cwd } = internalSlotsMap.get(this);
        const relativeTo = importerPath || path.join(cwd, "__placeholder__.js");
```
```
        const { cwd, resolvePluginsRelativeTo } = internalSlotsMap.get(this)
        const relativeTo = importerPath || path.join(resolvePluginsRelativeTo || cwd, "__placeholder__.js");
```

**Comment by mysticatea** [comment]
Hi.

The resolve**Plugins**RelativeTo option is for only plugins. Because ESLint loads plugins from the unique location in order to identity plugins by name.

The other things (custom parsers and shareable configs) are not so. You can use file paths in `extends` property. For example,

```js
module.exports = {
    extends: [
        require.resolve("path/to/airbnb-base"),
        "plugin:vue/essential"
    ]
}
```

**Comment by flyyang** [comment]
@mysticatea thanks for the reply. 

How about another option to solve this use case. Cause share config and plugins all using the same 
`extends` option, there is a little confuse when to require resolve, and when to use `resolvePluginsRelativeTo`.

**Comment by mysticatea** [comment]
We, the ESLint team, strongly recommend to not use global installations. Therefore, it is unlikely that functionality to encourage the use of global installation will be added.

`resolvePluginsRelativeTo` is an explicit name that means the option affects only plugins, and you can distinguish whether which are plugins or not by `plugin:` prefix.
